### ISSUE TITLE
warmup optimizer will get the same results in different epoch

### ISSUE THREAD

**Issue by sparkingarthur** [CyberZHG/keras-bert#91]
**Describe the Bug**
Actually,i'm trying to do earlystopping by myself, but i found that, every epoch get the same score, and I thought it's caused by the AdamWarmup optimizer (I changed the default Adam optimizer to ensure this, however, Adam got a much worse result than AdamWarmup)

**Minimal Codes To Reproduce**

```python
import keras_bert
from keras_bert import AdamWarmup, calc_train_steps
total_steps, warmup_steps = calc_train_steps(
    num_example=len(token_input),
    batch_size=32,
    epochs=1,
    warmup_proportion=0.05,
)
optimizer = AdamWarmup(total_steps, warmup_steps, lr=2e-5, min_lr=1e-5)
# optimizer = Adam(lr=2e-5)
model3.compile(loss=[focal_loss_fixed,'binary_crossentropy'], 
               loss_weights=[loss_weight, 1.0], 
               optimizer=optimizer,
               metrics=['acc']
              )
model3.summary()

##and training like this:
best = [-1, 0, 0, 0]  # socre, epoch, model.copy (if needed) , cv_result
earlystop = 2
model_bst_path = "../temp/keras-bert-best"
preds_name = "preds"

for epoch in range(1):
    model_path = "../temp/keras-bert-test_"+str(epoch)+".h5"
    model3.fit([token_input, seg_input, mask_input],[labels_train,labels_aux_train],batch_size=32,epochs=1)
    model3.save_weights(model_path)
    r = model3.predict([token_valid, seg_valid, mask_valid],verbose=1,batch_size=32)[0][:,0]
    s = evaluator.get_final_metric(r)
    print(epoch,s)
    if s > best[0]:# the bigger is better
        print("epoch " + str(epoch) + " improved from " + str(best[0]) + " to " + str(s))
        model_path = model_bst_path+".h5"
        model3.save_weights(model_path)
        best = [s,epoch,r]
    if epoch-best[1]>earlystop:
        break
```

**Comment by sparkingarthur** [comment]
BTW, thx a lot for the great contribution.
